### DISCUSSION TITLE
Command line arguments

### DISCUSSION THREAD

**Discussion by jgsuk** [sumatrapdfreader/sumatrapdf#2870] [General]
I have an application that records Product details, together with drawings (PDF files) associated with each Product.  In the application, I 'shell out' to Sumatra to print these drawings, and include the 'appropriate' command line arguments.

Some drawings are A4 and some are A3 (the file name contains the characters A3).  The printer (Toshiba e257) has two drawers, Drawer 1 loaded with A4 paper, and Drawer 2 with A3.  When I print, my command line includes:

for A4
-print-to "Toshiba e257" -print-settings "paper=A4,fit,bin=Drawer 1"

for A3
-print-to "Toshiba e257" -print-settings "paper=A3,fit,bin=Drawer 2"

Have I got my quotes correct, and do a need the paper= specification?

Also, I assume that  command line arguments will override setting held in the sumatra txt file.

Thanks,
Geoff

**Comment by GitHubRulesOK** [comment]
bins (drawers) may not always work as expected, so in such cases I would usually duplicate the driver as two extra named copies e.g.

Traditionally I would have 2 additional named drivers with fixed / saved settings (perhaps also advanced setting changed to `fit`)
ToshibaA4D1
ToshibaA3D2

thus no need for command line mods just `-print-to ToshibaA3D2 fileA3.pdf`

to verify drawer names can be tricky but see if Jens `PrinterInformation` tool  can verify correct names see link here  https://github.com/sumatrapdfreader/sumatrapdf/issues/1812#issuecomment-730617958

**Reply by jgsuk** [reply]
Thanks, that sounds a great idea.

This is in a networked environment, so is this a change that could be made on the server, or would all relevant workstations need updating? - excuse my ignorance, but this is something that I don't usually get involved with.

Regards,
Geoff

**Reply by GitHubRulesOK** [reply]
Without setting up a server to emulate yours I have no idea, but print servers are usually a single install (two extras in this case) then remote user should call the named \\\server printer (system rollout should be VBS scriptable)
Also perhaps explain to users not to change any user settings. One wrinkle is modern Windows 10/11 has a "Let windows manage my printers" switch/setting that can play havoc with the best set defaults.
